cache/cache.go

```diff
@@ -61,76 +63,97 @@ func New[T any](expireAfter time.Duration) *Cache[T] {
 		panic("expireAfter duration must be positive")
 	}
 
-	return &Cache[T]{
-		data:        make(map[string]item[T]),
+	c := &Cache[T]{
+		data:        make(map[string]*cacheListItem[T]),
 		expireAfter: expireAfter,
+		stopCh:      make(chan struct{}),
 	}
-}
-
-// Removes an item by name and expiry time when the purge was scheduled.
-// If there is a race, and the item has been refreshed, it will not be purged.
-func (c *Cache[T]) purgeExpired(name string, expectedExpiryTime int64) {
-	c.mu.Lock()
-	defer c.mu.Unlock()
 
-	if item, ok := c.data[name]; ok && item.expiresAt == expectedExpiryTime {
-		// found, and the expiry time is still the same as when the purge was requested.
-		delete(c.data, name)
+	// Start the sweep, with a minimum sweep of 50ms.
+	sweep := expireAfter / 4.0
```

Comment:
Why a quarter of expireAfter? I do understand why we would want to do sweeps more quickly than the expiry time, although i'm curious why specifically its a quarter. Is this just a number picked out of the air?

Reply:
It gives us a margin of error above 1/2, but doesn't put crazy pressure on the CPU for small expiration times.

Reply:
I recommend adding the "why" as a code comment, so someone years later knows the context and can make a change if needed.

Reply:
Commented inline